### PR TITLE
feat: added separate deployment for OpenShift and k8s

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
-  added separate deployment for OpenShift and k8s

### What issues does this PR fix or reference?
https://issues.redhat.com/projects/SVPI/issues/SVPI-30

- On Openshift used a build-in mechanism to generate certs for webhooks
- on K8s used cert-manager as a prerequisite.

### How to test this PR?
- on OpenShift `make deploy`
- on K8s `make install_cert_manager` , wait 10 sec , `make deploy_k8s `

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -190,6 +190,11 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
 	cd config/oauth && $(KUSTOMIZE) edit set image oauth=${SPIS_IMG} && cd ../..
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
 
+deploy_k8s: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${SPIO_IMG} && cd ../..
+	cd config/oauth && $(KUSTOMIZE) edit set image oauth=${SPIS_IMG} && cd ../..
+	$(KUSTOMIZE) build config/k8s | kubectl apply -f -
+
 undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
```
Comment (metlos) [inline Makefile:198]: Shouldn't there be a dedicated `undeploy_k8s`, too?

File: config/webhook/service.yaml [hunk 2]
```diff
@@ -4,6 +4,8 @@ kind: Service
 metadata:
   name: webhook-service
   namespace: system
+  annotations:
+    service.beta.openshift.io/serving-cert-secret-name: webhook-server-cert
```
Comment (metlos) [inline config/webhook/service.yaml:8]: This secret is currently generated by the certmanager. Where do you generate it in openshift deployment?

Comment (skabashnyuk) [reply config/webhook/service.yaml:8]: that is an OpenShift feature that will generate that secret.
https://docs.openshift.com/container-platform/4.6/security/certificates/service-serving-certificate.html#add-service-certificate_service-serving-certificate

File: Makefile (hunk 1 above)
Comment (skabashnyuk) [reply Makefile:201]: done

**Review by metlos — APPROVED** [review]

**Pull request merged**